src/classes/Hotel.js

```diff
@@ -0,0 +1,51 @@
+class Hotel {
+  constructor(usersData, roomsData, bookingsData) {
+    this.guests = usersData || [];
+    this.rooms = roomsData || [];
+    this.bookings = bookingsData || [];
+    this.availableRooms = [];
+    this.filterTerm = '';
+  }
+
+  setAvailableRooms(date) {
+    const bookedRooms = this.bookings.reduce((acc, booking) => {
+      this.rooms.forEach(room => {
+        if (booking.date === date && room.number === booking.roomNumber) {
+          acc.push(room);
+        }
+      })
+      return acc;
+    }, []);
+
+    this.rooms.forEach(room => {
+      if (!bookedRooms.includes(room)) {
+        this.availableRooms.push(room);
+      }
+    })
+  }
+
+  checkAvailableRoomsByDate(date) {
+    const todaysDate = new Date().toISOString().split('T')[0].split('-').join('/');
```

Comment:
I'll have to look at this toISOString method. Well done getting the date how you want it.